File: frontend/src/api/samplePool.ts
import api from './axios'

export interface AddHardRequest {
  dataset_id: string
  model_id: string
  image_names: string[]
  version?: string
}

export interface AddBackgroundRequest {
  dataset_id: string
  image_names: string[]
  version?: string
}

export interface HardPoolInfo {
  model_id: string
  model_code?: string
  names: string[]
  image_count: number
  label_count: number
  updated_at: string
}

export interface SamplePoolResult {
  hard: HardPoolInfo[]
  background: number
}

/** 将某数据集的指定图片（连同标注）加入指定模型的困难样本库 */
export const addHardSamples = async (params: AddHardRequest) => {
  const { data } = await api.post('/sample-pool/hard', params)
  return data
}

/** 将某数据集的指定图片作为无目标背景加入空白样本库 */
export const addBackgroundSamples = async (params: AddBackgroundRequest) => {
  const { data } = await api.post('/sample-pool/background', params)
  return data
}

/** 汇总样本池状态：困难样本（按模型）+ 空白样本（全局） */
export const listSamplePool = async (): Promise<SamplePoolResult> => {
  const { data } = await api.get('/sample-pool')
  return data
}

/** 清空某模型的困难样本库 */
export const clearHardPool = async (modelId: string) => {
  const { data } = await api.delete(`/sample-pool/hard/${modelId}`)
  return data
}

/** 清空空白样本库 */
export const clearBackgroundPool = async () => {
  const { data } = await api.delete('/sample-pool/background')
  return data
}

File: backend/src/api/routes/sample_pool.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import Optional, List
from src.services.sample_pool_service import SamplePoolService

router = APIRouter(prefix="/sample-pool", tags=["sample-pool"])

sample_pool_service = SamplePoolService()


class AddHardRequest(BaseModel):
    dataset_id: str
    model_id: str
    image_names: List[str]
    version: Optional[str] = "v1"


class AddBackgroundRequest(BaseModel):
    dataset_id: str
    image_names: List[str]
    version: Optional[str] = "v1"


@router.post("/hard")
async def add_hard_samples(req: AddHardRequest):
    """把数据集指定图片（连同标注）加入某模型的困难样本库（1.6）"""
    try:
        return await sample_pool_service.add_hard_samples(
            req.dataset_id, req.model_id, req.image_names, req.version
        )
    except ValueError as e:
        raise HTTPException(400, str(e))


@router.post("/background")
async def add_background_samples(req: AddBackgroundRequest):
    """把数据集指定图片作为无目标背景（负样本）加入全局空白样本库（1.6）"""
    try:
        return await sample_pool_service.add_background_samples(
            req.dataset_id, req.image_names, req.version
        )
    except ValueError as e:
        raise HTTPException(400, str(e))


@router.get("")
async def list_pool():
    """样本池状态：困难样本（按模型）+ 空白样本（全局）"""
    return await sample_pool_service.list_pool()


@router.delete("/hard/{model_id}")
async def clear_hard_pool(model_id: str):
    """清空某模型的困难样本库"""
    return await sample_pool_service.clear_hard(model_id)


@router.delete("/background")
async def clear_background_pool():
    """清空全局空白样本库"""
    return await sample_pool_service.clear_background()